lib/usd/utils/util.cpp

```diff
@@ -171,8 +171,14 @@ updateInternalReferences(const UsdPrim& oldPrim, const UsdPrim& newPrim)
                             return false;
                         }
 
+                        // NOTE: currently there is no APIs either at 
+                        // Sdf or Usd level to help querying the "append" or 
+                        // "prepend" operations for the entries in the list. Therefore, UsdListPosition 
+                        // is always set by the default value UsdListPositionBackOfPrependList passed to
+                        // AddInternalReference
+
                         // add the new internal reference
-                        status = p.GetReferences().AddInternalReference(newPath);
+                        status = p.GetReferences().AddInternalReference(finalPath);
```

Comment:
currently there is no APIs either at Sdf or Usd level to help querying the "append" or "prepend" operations for the entries in the list. Therefore, UsdListPosition is always set by the default value UsdListPositionBackOfPrependList passed to AddInternalReference.

This needs to be documented.